x-pack/plugins/security_solution/public/timelines/components/flyout/action_menu/new_timeline.tsx

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EuiButtonEmpty, EuiFlexGroup, EuiFlexItem, EuiPopover } from '@elastic/eui';
+import React, { useMemo, useState, useCallback } from 'react';
+import { NewTimeline } from '../../timeline/properties/helpers';
+import { NewTemplateTimeline } from '../../timeline/properties/new_template_timeline';
+import * as i18n from './translations';
+
+interface NewTimelineActionProps {
+  timelineId: string;
+}
+
+export const NewTimelineAction = ({ timelineId }: NewTimelineActionProps) => {
+  const [isPopoverOpen, setPopover] = useState(false);
+
+  const closePopover = useCallback(() => {
+    setPopover(false);
+  }, []);
+
+  const togglePopover = useCallback(() => setPopover((prev) => !prev), []);
+
+  const onActionBtnClick = useCallback(() => {
+    togglePopover();
+  }, [togglePopover]);
+
+  const newTimelineActionbtn = useMemo(() => {
+    return (
+      <EuiButtonEmpty iconType="arrowDown" size="s" iconSide="right" onClick={onActionBtnClick}>
+        {i18n.NEW_TIMELINE_BTN}
+      </EuiButtonEmpty>
+    );
+  }, [onActionBtnClick]);
+
+  return (
+    <EuiPopover
+      button={newTimelineActionbtn}
+      isOpen={isPopoverOpen}
+      closePopover={closePopover}
+      panelStyle={{
```

Comment:
You should pull this object out of the component so it doesn't get recreated on every render. No reason to introduce useMemo since it's essentially a constant so..

```
const panelStyle = { padding: 0 };

export const NewTimelineAction = () => {... <EuiPopover panelStyle={panelStyle}....> }
```
```

Reply:
@michaelolo24 , this is done as well.